src/delete/usecases/error.ts

```diff
@@ -0,0 +1,48 @@
+import { KintoneAllRecordsError } from "@kintone/rest-api-client";
+import type { KintoneRecordForDeleteAllParameter } from "../../kintone/types";
+
+export class DeleteAllRecordsError extends Error {
+  private readonly cause: unknown;
+  private readonly records: KintoneRecordForDeleteAllParameter[];
+  private readonly numOfSuccess: number;
+  private readonly numOfTotal: number;
+
+  constructor(cause: unknown, records: KintoneRecordForDeleteAllParameter[]) {
+    const message = "Failed to delete all records.";
+    super(message);
+
+    this.name = "DeleteAllRecordsError";
+    this.message = message;
+    this.cause = cause;
+    this.records = records;
+    this.numOfTotal = this.records.length;
+    this.numOfSuccess = 0;
+    if (this.cause instanceof KintoneAllRecordsError) {
+      this.numOfSuccess = this.cause.numOfProcessedRecords;
+    }
+
+    // https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    // Set the prototype explicitly.
+    Object.setPrototypeOf(this, DeleteAllRecordsError.prototype);
+  }
+
+  toString(): string {
+    let errorMessage = "";
+    errorMessage += this.message + "\n";
+
+    if (this.numOfSuccess === 0) {
+      errorMessage += `No records are deleted.\n`;
+    } else {
+      errorMessage += `${this.numOfSuccess}/${this.numOfTotal} records are deleted successfully.\n`;
+    }
+
+    if (this.cause instanceof KintoneAllRecordsError) {
+      errorMessage += `An error occurred while deleting all records.\n${this.cause.error.message}`;
```

Comment:
Why don't we use `kintoneAllRecordsErrorToString()`?

It would be nice to fix the `kintoneAllRecordsErrorToString()` to make it can be used in DeleteAllRecordsError.

Reply:
@mshrtsr 

> Why don't we use kintoneAllRecordsErrorToString()

I see that the almost logic of `kintoneAllRecordsErrorToString()` is redundant for deleting all records.
Because when an error occurs, the `KintoneAllRecordsError.error.errors` is always `undefined`, and only `KintoneAllRecordsError.error.message` is available.
(I checked a test case which has wrong `records[].revision` and `records[].id`)
==> There is different behavior for importing and deleting all records
==> For deleting all records, we only need the info of `KintoneAllRecordsError.error.message`. That's why I don't use `kintoneAllRecordsErrorToString`

However, anyway, all of them are `KintoneAllRecordsError`, so I will try to unify them, but they still are easy for modifying when extending

Reply:
I saw you abstracted the parsing process of KintoneAllRecordsError.
It looks nice!